azkaban-common/src/main/java/azkaban/jobExecutor/AbstractProcessJob.java

```diff
@@ -81,6 +84,31 @@ protected void resolveProps() {
     jobProps = PropsUtils.resolveProps(jobProps);
   }
 
+  /**
+   * prints the current Job props to the Job log.
+   */
+  protected void logJobProperties() {
+    if (this.jobProps != null &&
+        this.jobProps.getBoolean(JOB_DUMP_PROPERTIES_IN_LOG, false)){
+      try {
+        Map<String,String> flattenedProps = this.jobProps.getFlattened();
```

Comment:
I think there's merits going both ways, so providing a flatten method is probably also fine

if we are going the treeset route then yes, we'll still need to new a TreeSet...